service/app/queries/published_log_test.go

```diff
@@ -0,0 +1,68 @@
+package queries_test
+
+import (
+	"testing"
+
+	"github.com/planetary-social/scuttlego/di"
+	"github.com/planetary-social/scuttlego/fixtures"
+	"github.com/planetary-social/scuttlego/internal"
+	"github.com/planetary-social/scuttlego/service/app/queries"
+	"github.com/planetary-social/scuttlego/service/domain/refs"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPublishedLog_NilStarSeqIsNotUsedToDetermineMessageSequence(t *testing.T) {
+	app, err := di.BuildTestQueries()
+	require.NoError(t, err)
+
+	query := queries.PublishedLog{
+		StartSeq: nil,
+	}
+
+	_, err = app.Queries.PublishedLog.Handle(query)
+	require.NoError(t, err)
+
+	require.Empty(t, app.ReceiveLogRepository.GetMessageCalls)
+}
+
+func TestPublishedLog_NotNilStarSeqIsUsedToDetermineMessageSequence(t *testing.T) {
+	app, err := di.BuildTestQueries()
+	require.NoError(t, err)
+
+	localFeed := refs.MustNewIdentityFromPublic(app.LocalIdentity).MainFeed()
+
+	seq := fixtures.SomeReceiveLogSequence()
+	msg := fixtures.SomeMessage(fixtures.SomeSequence(), localFeed)
+
+	query := queries.PublishedLog{
+		StartSeq: internal.Ptr(seq),
+	}
+
+	app.ReceiveLogRepository.MockMessage(seq, msg)
+
+	_, err = app.Queries.PublishedLog.Handle(query)
+	require.NoError(t, err)
+
+	require.NotEmpty(t, app.ReceiveLogRepository.GetMessageCalls)
```

Comment:
Maybe we could check that the exact `StartSeq` was passed to the repository?

Reply:
That is a definite improvement to those tests. [d13024a](https://github.com/planetary-social/scuttlego/pull/28/commits/d13024a0eaa699b51b9e5c27fbf268b05a92f7ee)